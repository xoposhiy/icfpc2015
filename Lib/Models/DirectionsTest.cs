using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.CompilerServices;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace Lib.Models
{
    [TestFixture]
    public class DirectionsTest
    {
        [MethodImpl(MethodImplOptions.NoInlining)]
        [Test, UseReporter(typeof(DiffReporter))]
        public void Test()
        {
            var directions = Phrases.all.Select(w => w + " → " + string.Join(" ", w.ToDirections().Select(d => d.ToString())));
            Console.WriteLine(directions);
            Approvals.VerifyAll(directions, "d");
        }
    }
}