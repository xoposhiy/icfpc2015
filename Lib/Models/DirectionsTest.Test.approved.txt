d[0] = Ei! → E SW W
d[1] = R'lyeh → CW W SE E E SW
d[2] = Ia! Ia! → SW SW W SE SW SW W
d[3] = Yuggoth → E CCW SW SW SE CCW SW
d[4] = YogSothoth → E SE SW CCW SE CCW SW SE CCW SW
d[5] = blue hades → E SE CCW E SE SW SW CW E CCW
d[6] = Tsathoggua → CCW CCW SW CCW SW SE SW SW CCW SW
d[7] = Necronomicon → SE E E CW SE SE SE SE SW E SE SE
d[8] = cthulhu fhtagn! → E CCW SW CCW SE SW CCW SE E SW CCW SW SW SE W
d[9] = Ph'nglui mglw'nafh Cthulhu R'lyeh wgah'nagl fhtagn. → W SW W SE SW SE CCW SW SE SE SW SE CCW W SE SW E SW SE E CCW SW CCW SE SW CCW SE CW W SE E E SW SE CCW SW SW SW W SE SW SW SE SE E SW CCW SW SW SE W
